### PR TITLE
chore(application-templates): Lifecycle and Delete overhaul

### PR DESCRIPTION
**Pull request by obmagnusson**
## What

Removed the DEPRECATED_DefaultLifecycle. Every application state updated with the DefaultLifecycle (30 days). Delete : true set where applicable.

## Why

Get every application template MÁP compliant. 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by albinagu — APPROVED** [review]
looks good, but a question on whether delete should be permitted for assignee and after payment

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts [hunk 1]
```diff
@@ -128,6 +131,7 @@ const MarriageConditionsTemplate: ApplicationTemplate<
                   Promise.resolve(val.done),
                 ),
               read: 'all',
+              delete: true,
```
Comment (albinagu) [inline libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:134]: here, the applicant has already paid for the application, and assignee about to confirm. Does it make sense to allow to delete the application at that state?

**Review by Thorrik58 — COMMENTED** [review]
Remove unused then we are good

**Review by johannaagma — APPROVED** [review]

**Review by HelgiHH — APPROVED** [review]

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts (hunk 1 above)
Comment (obmagnusson) [reply libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:134]: ill remove it

File: libs/application/templates/mortgage-certificate/src/lib/mortgageCertificateTemplate.ts [hunk 2]
```diff
@@ -77,6 +77,7 @@ const template: ApplicationTemplate<
                 },
               ],
               write: 'all',
+              delete: true,
```
[the remark this answers is not in the document]
Comment (obmagnusson) [reply libs/application/templates/mortgage-certificate/src/lib/mortgageCertificateTemplate.ts:80]: No it does not .

**Review by davidpxn — APPROVED** [review]

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 3]
```diff
@@ -170,7 +170,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.otherParentApprovalDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:173]: The application has to be available in at least 2 years and 8 months.
Is it? pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 4]
```diff
@@ -233,7 +233,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.otherParentActionDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:236]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 5]
```diff
@@ -264,7 +264,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.employerWaitingToAssignDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:267]: Same above: pruneAfterDays(360) // available for 1 year

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 6]
```diff
@@ -296,7 +296,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.employerApprovalDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:299]: Same above: pruneAfterDays(360) // available for 1 year

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 7]
```diff
@@ -354,7 +354,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.employerActionDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:357]: Same above: pruneAfterDays(360) // available for 1 year

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 8]
```diff
@@ -386,7 +386,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.vinnumalastofnunApprovalDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:389]: The application has to be available in at least 2 years and 8 months here
Is it? pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 9]
```diff
@@ -424,7 +424,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.vinnumalastofnunActionDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:427]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 10]
```diff
@@ -448,7 +448,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.approvedDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:451]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 11]
```diff
@@ -475,7 +475,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.editOrAddPeriodsDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:478]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 12]
```diff
@@ -514,7 +514,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
             description:
               statesMessages.employerWaitingToAssignForEditsDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:517]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 13]
```diff
@@ -544,7 +544,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.employerApproveEditsDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:547]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 14]
```diff
@@ -574,7 +574,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.employerEditsActionDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:577]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 15]
```diff
@@ -602,7 +602,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.vinnumalastofnunApproveEditsDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:605]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 16]
```diff
@@ -630,7 +630,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           actionCard: {
             description: statesMessages.vinnumalastofnunEditsActionDescription,
           },
-          lifecycle: DEPRECATED_DefaultStateLifeCycle,
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:633]: Same above: pruneAfterDays(970) // available for 2 years and 8 months

**Review by qdd1 — APPROVED** [review]
LGTM

**Review by unnarb — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**